egs/wsj/s5/steps/nnet3/multilingual/allocate_multilingual_examples.py

```diff
@@ -0,0 +1,310 @@
+                    break
+            all_egs.append(this_egs)
+            archfile.close()
+
+    logger.info("combine examples across all jobs correspond for each archive.")
+    for archive in range(num_archives):
+        logger.info("Combine egs.job.{0}.scp across all jobs into "
+                    "egs.{0}.scp.".format(archive))
+        this_ranges = []
+        f = open("{0}/temp/{1}ranges.{2}".format(
+                 args.egs_dir, args.prefix, archive + 1), 'w')
+        o = open("{0}/{1}output.{2}.ark".format(
+                args.egs_dir, args.prefix, archive + 1), 'w')
+        w = open("{0}/{1}weight.{2}.ark".format(
+                args.egs_dir, args.prefix, archive + 1), 'w')
+        scp_per_archive_file = open("{0}/{1}egs.{2}.scp"
+                                    "".format(args.egs_dir,
+                                              args.prefix, archive + 1), 'w')
+
+        # check files before writing.
+        if f is None:
+            raise Exception("Error opening file {0}".format(f))
+        if o is None:
+            raise Exception("Error opening file {0}".format(o))
+        if w is None:
+            raise Exception("Error opening file {0}".format(w))
+        if scp_per_archive_file is None:
+            raise Exception("Error opening file {0}".format(scp_per_archive_file))
+
+        for job in range(args.num_jobs):
+            scp = ("{0}/temp/{1}scp.{2}.{3}".format(args.egs_dir, args.prefix,
+                                                    job + 1, archive + 1))
+            with open(scp,"r") as scpfile:
+                for line in scpfile:
+                    scp_line = line.splitlines()[0].split()
+                    print("{0} {1}".format(scp_line[0], scp_line[1]),
+                                           file=scp_per_archive_file)
+                    print("{0} output-{1}".format(scp_line[0], scp_line[2]),
+                                                  file=o)
+                    print("{0} {1}".format(
+                        scp_line[0],
+                        lang2weight[int(scp_line[2])]), file=w)
+            os.remove(scp)
+
+        for (lang_id, start_eg_line, num_egs) in all_egs[num_archives * job + archive]:
+            this_ranges.append((lang_id, start_eg_line, num_egs))
+
+        # write ranges.archive
+        for (lang_id, start_eg_line, num_egs) in this_ranges:
+            print("{0} {1} {2}".format(lang_id, start_eg_line, num_egs), file=f)
+
+        f.close()
+        o.close()
+        w.close()
+        scp_per_archive_file.close()
+    logger.info("allocate_multilingual_examples.py finished generating "
+                "{0}egs.*.scp, {0}.ranges.*, {0}output.*.ark "
+                "and {0}weights.*.ark files".format(args.prefix))
+def main():
+    try:
+        args = get_args()
+        process_multilingual_egs(args)
+    except Exception as e:
+        traceback.print_exc()
+        sys.exit(1)
+
+if __name__ == "__main__":
```

Comment:
@vimalmanohar, I don't have time to go through this file in detail; would you mind checking it?